packages/MarkdownEditor-Tests.package/MarkdownHeadingsTest.class/instance/testConvertingToTextUsesInlineParser.st

```diff
@@ -0,0 +1,12 @@
+tests
+testConvertingToTextUsesInlineParser
+
+	| markdownString mockInlineParser heading result |
+	markdownString := '# **Stronger** heading with *emphasis*'.
+	mockInlineParser := MarkdownMockInlineParser new.
+	heading := MarkdownHeading newFrom: markdownString.
+	heading inlineParser: mockInlineParser.
+	result := heading asText.
+	self
+		assert: mockInlineParser markerText equals: result;
```

Comment:
If I understand correctly here we only want to test if the `mockInlineParser` as received the message `parse:`. Why do we do this by comparing two strings? Couldn't the `mockInlineParser` have a Boolean variable that starts as `false` and gets set to `true` once it has received the message `parse:`? Then we could ask it `hasBeenAskedToParse` or something like that and then this line could be replaced with `self assert: mockInlineParser hasBeenAskedToParse`. 

Or even simpler: Line 12 already asserts that the `mockInlineParser ` remembers the String. It can only remember the String if it has been asked to parse the String. Thus why do we even have to test if it has been asked to parse when we also test that it memorizes the correct String, which it only can if it has been asked to parse it?

Reply:
+1 for the Boolean version.

Reply:
Actually, I intented to test two things:

1. The result of the inline parser is actually returned (as in, the block parser not just runs it and then forgets about the results)
2. The inline parser was fed with the correct string

The second assert is easier to understand and can be implemented in multiple ways (which is requested by you). The first assert, if the result is actually returned untouched is done by mocking the inline parsing as if it would always return a magic value, `markerText`. As we receive the `markerText`, we can be sure that the inline parsing was passed down to the text-converting.

Imo this makes the test valuable, because the block parser is asserted to follow the correct path of parsing. So, I see two options: We further improve the test, then I'll need more information, what you don't understand, where a comment or other code structures would be helpful. We remove the first assert, because it adds only little value for you (the option I obviously would not prefer).

Regarding your off-topic request: No, `cr` as instance variable is not a good idea, because it makes the string un-inspectable (at least in my two-minute class). Yes, there is bloat, and I would be happy to see a more compact solution with good semantics.

Reply:
Okay I understand now. 

The problem is, that it is not clear that the `mockInlineParser` always returns the marker String. Thus one doesn't understand that after line 9 the `result` should be the marker string of the `mockInlineParser` and because of that line 11 doesn't make sense.

I think it would be easier to understand if one could set the string that is always returned by the `MockInlineParser`. Then that would have to happen in the test using a method like `parseResult: aText` and then you could assure that `result` equals `aText`. 

This would eliminate the markerString which is very confusing. Also you would see that the `MockInlineParser` has a fixed parsing result which would make it easy to understand why `result` should be equal to the `parseResult`.

Reply:
> The problem is, that it is not clear that the `mockInlineParser` always returns the marker String.

This should (now) be pretty obvious after looking into the source of the mock object (It belongs to the test case, so I think I can expect the reader to know how it works. It's really simple, so this should be not a big problem).

I'd prefer to rename the name of `markerText` to `parsingResult` or similiar. Your proposed change looks like you have a problem with the semantics of the `markerText` which also can be addressed by renaming the variable instead of micro-managing the mock object

Reply:
Renaming the `markerText` to `parsingResult` would already be a huge improvement. The only problem I have with that would be that the String which is returned by `parsingResult` comes out of nowhere. I personally would say that that is weird and kind of ugly but since that is such a small  detail it won't hinder my approval. Maybe the others (@lpfennigschmidt, @kiragrammel, @felix-gohla) could comment which version they would prefer.

Reply:
Renamed to `parsingResult` for now. We can discuss this further in the future